mint-auditor/src/bin/main.rs

```diff
@@ -170,32 +185,49 @@ fn cmd_get_block_audit_data(
     json: bool,
     logger: Logger,
 ) {
-    let mint_auditor_db = MintAuditorDb::open(&mint_auditor_db_path, logger.clone())
-        .expect("Could not open mint auditor DB");
-
-    let block_index = block_index
-        .or_else(|| {
-            mint_auditor_db
-                .last_synced_block_index()
-                .expect("Could not get last synced block index")
-        })
-        .expect("No blocks synced");
-
-    let audit_data = mint_auditor_db
-        .get_block_audit_data(block_index)
-        .expect("Could not get audit data for block");
-
-    if json {
-        println!(
-            "{}",
-            serde_json::to_string(&audit_data).expect("failed serializing json")
-        );
-    } else {
-        println!("Block index: {}", block_index);
-        for (token_id, balance) in audit_data.balance_map.iter() {
-            println!("Token {}: {}", token_id, balance);
+    let mint_auditor_db = MintAuditorDb::new_from_path(
+        &mint_auditor_db_path.into_os_string().into_string().unwrap(),
+        DB_POOL_SIZE,
+        logger.clone(),
+    )
+    .expect("Could not open mint auditor DB");
+
+    let conn = mint_auditor_db
+        .get_conn()
+        .expect("Could not get db connection");
+
+    transaction(&conn, |conn| -> Result<(), Error> {
+        let block_index = block_index
+            .or_else(|| {
+                BlockAuditData::last_synced_block_index(conn)
+                    .expect("Could not get last synced block index")
+            })
+            .expect("No blocks synced");
+
+        let audit_data =
+            BlockAuditData::get(conn, block_index).expect("Could not get audit data for block");
+        let balance_map = BlockBalance::get_balances_for_block(conn, block_index)
+            .expect("Could not get balances for block");
```

Comment:
Is it the right strategy to panic con a connection? Does that cleanup the connection pool properly?

Reply:
I don't think there's any concern in this panicing if it fails reading from the database. The connection pool only lives inside the process so that will go away together with everything else. But to make things easier to reason about I'll change this to return a Result<> and panic after the db is dropped.